### ISSUE TITLE
Self-signed certificates: ```curl: (35) OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to API_IP:443```

### ISSUE THREAD

**Issue by i2rulloa** [docker/for-mac#5524]
- [X] I have tried with the latest version of Docker Desktop
  - [X] I have tried disabling enabled experimental features
  - [X] I have uploaded Diagnostics
  - Diagnostics ID: 38E55D99-754A-49D0-AA69-B7E137F4729D/20210331135025

### Expected behavior
Inside a docker container with debian:10.7 for example
Tried to `kubectl get node` or `curl -k https://API` and get response.

### Actual behavior
Inside a docker container with debian:10.7 for example
Tried to `kubectl get node` or `curl -k https://API_IP` and none of them works with my GKE endpoint.

With kubectl: `kubectl -v=8 get node`

```
I0331 16:20:27.114543      41 loader.go:379] Config loaded from file:  /home/xxx/.kube/config
I0331 16:20:27.120687      41 round_trippers.go:422] GET https://API_IP/api?timeout=32s
I0331 16:20:27.120705      41 round_trippers.go:429] Request Headers:
I0331 16:20:27.120709      41 round_trippers.go:433]     Accept: application/json, */*
I0331 16:20:27.120733      41 round_trippers.go:433]     User-Agent: kubectl/v1.20.1 (linux/arm64) kubernetes/c4d7527
I0331 16:20:32.919404      41 loader.go:379] Config loaded from file:  /home/xxx/.kube/config
I0331 16:20:32.934352      41 round_trippers.go:448] Response Status:  in 5813 milliseconds
I0331 16:20:32.934539      41 round_trippers.go:451] Response Headers:
I0331 16:20:32.934577      41 request.go:943] Got a Retry-After 1s response for attempt 1 to https://API_IP/api?timeout=32s
```

With curl: `curl -vvv -k https://API_IP` tell me this.
```
*   Trying API_IP:443...
* Connected to API_IP (API_IP) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: none
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to API_IP:443
* Closing connection 0
curl: (35) OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to API_IP:443
```

### Information
Please, help us understand the problem.  For instance:
  - Is it reproducible? No, since I do not want to tell my endpoint.
  - Is the problem new? Yes
  - Did the problem appear with an update? Yes

  - macOS Version: 11.1 (20C69)
  - Docker Desktop Version: 3.3.0 (62345) - Apple Silicon Tech Preview RC 2

### Steps to reproduce the behavior

GKE endpoint has a certificate (automatically generated) that is not trusted by all browsers and will expire in nearly 5 years.

I have tried with ```curlimages/curl:latest``` image, also add ```--tls-max 1.2``` to the curl command and always same issue.
I have tried this in my mac shell and it works fine.

It seems to me that something changes in the latest upgrade.

**Comment by stellirin** [comment]
This is not only M1, My 2019 MacBook Pro with Intel i9 has the same issue connecting to GKE endpoints.

I first tried different container images (ubuntu:focal, ubuntu:bionic, ubuntu:xenial, debian:buster) and differnet kubectl versions in the containers (1.21.0, 1.19.7).

The 1ms response times strongly suggests its being blocked by Docker.

```
kubectl --v=10 version
I0413 11:09:03.350011    4745 loader.go:375] Config loaded from file:  /home/vscode/.kube/config
I0413 11:09:03.351015    4745 round_trippers.go:424] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.19.7 (linux/amd64) kubernetes/1dd5338" 'https://w.x.y.z/version?timeout=32s'
I0413 11:09:03.352696    4745 round_trippers.go:444] GET https://w.x.y.z/version?timeout=32s  in 1 milliseconds
I0413 11:09:03.352757    4745 round_trippers.go:450] Response Headers:
I0413 11:09:03.352806    4745 request.go:933] Got a Retry-After 1s response for attempt 1 to https://w.x.y.z/version?timeout=32s
I0413 11:09:04.353696    4745 round_trippers.go:424] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.19.7 (linux/amd64) kubernetes/1dd5338" 'https://w.x.y.z/version?timeout=32s'
I0413 11:09:04.357126    4745 round_trippers.go:444] GET https://w.x.y.z/version?timeout=32s  in 3 milliseconds
I0413 11:09:04.357148    4745 round_trippers.go:450] Response Headers:
I0413 11:09:04.357179    4745 request.go:933] Got a Retry-After 1s response for attempt 2 to https://w.x.y.z/version?timeout=32s
I0413 11:09:05.357868    4745 round_trippers.go:424] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.19.7 (linux/amd64) kubernetes/1dd5338" 'https://w.x.y.z/version?timeout=32s'
I0413 11:09:05.360473    4745 round_trippers.go:444] GET https://w.x.y.z/version?timeout=32s  in 2 milliseconds
I0413 11:09:05.360493    4745 round_trippers.go:450] Response Headers:
I0413 11:09:05.360520    4745 request.go:933] Got a Retry-After 1s response for attempt 3 to https://w.x.y.z/version?timeout=32s
I0413 11:09:06.363369    4745 round_trippers.go:424] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.19.7 (linux/amd64) kubernetes/1dd5338" 'https://w.x.y.z/version?timeout=32s'
I0413 11:09:06.366231    4745 round_trippers.go:444] GET https://w.x.y.z/version?timeout=32s  in 2 milliseconds
I0413 11:09:06.366289    4745 round_trippers.go:450] Response Headers:
I0413 11:09:06.366322    4745 request.go:933] Got a Retry-After 1s response for attempt 4 to https://w.x.y.z/version?timeout=32s
I0413 11:09:07.367419    4745 round_trippers.go:424] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.19.7 (linux/amd64) kubernetes/1dd5338" 'https://w.x.y.z/version?timeout=32s'
I0413 11:09:07.370371    4745 round_trippers.go:444] GET https://w.x.y.z/version?timeout=32s  in 2 milliseconds
I0413 11:09:07.370440    4745 round_trippers.go:450] Response Headers:
I0413 11:09:07.370549    4745 request.go:933] Got a Retry-After 1s response for attempt 5 to https://w.x.y.z/version?timeout=32s
I0413 11:09:08.373385    4745 round_trippers.go:424] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.19.7 (linux/amd64) kubernetes/1dd5338" 'https://w.x.y.z/version?timeout=32s'
I0413 11:09:08.376631    4745 round_trippers.go:444] GET https://w.x.y.z/version?timeout=32s  in 3 milliseconds
I0413 11:09:08.376680    4745 round_trippers.go:450] Response Headers:
I0413 11:09:08.376711    4745 request.go:933] Got a Retry-After 1s response for attempt 6 to https://w.x.y.z/version?timeout=32s
I0413 11:09:09.380902    4745 round_trippers.go:424] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.19.7 (linux/amd64) kubernetes/1dd5338" 'https://w.x.y.z/version?timeout=32s'
I0413 11:09:09.384785    4745 round_trippers.go:444] GET https://w.x.y.z/version?timeout=32s  in 3 milliseconds
I0413 11:09:09.384879    4745 round_trippers.go:450] Response Headers:
I0413 11:09:09.384947    4745 request.go:933] Got a Retry-After 1s response for attempt 7 to https://w.x.y.z/version?timeout=32s
I0413 11:09:10.389703    4745 round_trippers.go:424] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.19.7 (linux/amd64) kubernetes/1dd5338" 'https://w.x.y.z/version?timeout=32s'
I0413 11:09:10.392512    4745 round_trippers.go:444] GET https://w.x.y.z/version?timeout=32s  in 2 milliseconds
I0413 11:09:10.392567    4745 round_trippers.go:450] Response Headers:
I0413 11:09:10.392611    4745 request.go:933] Got a Retry-After 1s response for attempt 8 to https://w.x.y.z/version?timeout=32s
I0413 11:09:11.393877    4745 round_trippers.go:424] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.19.7 (linux/amd64) kubernetes/1dd5338" 'https://w.x.y.z/version?timeout=32s'
I0413 11:09:11.395613    4745 round_trippers.go:444] GET https://w.x.y.z/version?timeout=32s  in 1 milliseconds
I0413 11:09:11.395647    4745 round_trippers.go:450] Response Headers:
I0413 11:09:11.395716    4745 request.go:933] Got a Retry-After 1s response for attempt 9 to https://w.x.y.z/version?timeout=32s
I0413 11:09:12.397099    4745 round_trippers.go:424] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.19.7 (linux/amd64) kubernetes/1dd5338" 'https://w.x.y.z/version?timeout=32s'
I0413 11:09:12.400288    4745 round_trippers.go:444] GET https://w.x.y.z/version?timeout=32s  in 2 milliseconds
I0413 11:09:12.400338    4745 round_trippers.go:450] Response Headers:
I0413 11:09:12.400483    4745 request.go:933] Got a Retry-After 1s response for attempt 10 to https://w.x.y.z/version?timeout=32s
I0413 11:09:13.400995    4745 round_trippers.go:424] curl -k -v -XGET  -H "User-Agent: kubectl/v1.19.7 (linux/amd64) kubernetes/1dd5338" -H "Accept: application/json, */*" 'https://w.x.y.z/version?timeout=32s'
I0413 11:09:13.402920    4745 round_trippers.go:444] GET https://w.x.y.z/version?timeout=32s  in 1 milliseconds
I0413 11:09:13.402986    4745 round_trippers.go:450] Response Headers:
I0413 11:09:13.403022    4745 request.go:1097] Response Body: 
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.7", GitCommit:"1dd5338295409edcfff11505e7bb246f0d325d15", GitTreeState:"clean", BuildDate:"2021-01-13T13:23:52Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
I0413 11:09:13.403225    4745 helpers.go:216] server response object: [{
  "metadata": {},
  "status": "Failure",
  "message": "an error on the server (\"\") has prevented the request from succeeding",
  "reason": "InternalError",
  "details": {
    "causes": [
      {
        "reason": "UnexpectedServerResponse"
      }
    ],
    "retryAfterSeconds": 1
  },
  "code": 500
}]
F0413 11:09:13.403307    4745 helpers.go:115] Error from server (InternalError): an error on the server ("") has prevented the request from succeeding
goroutine 1 [running]:
k8s.io/kubernetes/vendor/k8s.io/klog/v2.stacks(0xc00000e001, 0xc0001a91d0, 0x97, 0xe8)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/klog/v2/klog.go:996 +0xb9
k8s.io/kubernetes/vendor/k8s.io/klog/v2.(*loggingT).output(0x2d04b80, 0xc000000003, 0x0, 0x0, 0xc00018d030, 0x2ae4041, 0xa, 0x73, 0x40b300)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/klog/v2/klog.go:945 +0x191
k8s.io/kubernetes/vendor/k8s.io/klog/v2.(*loggingT).printDepth(0x2d04b80, 0x3, 0x0, 0x0, 0x2, 0xc00087daf0, 0x1, 0x1)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/klog/v2/klog.go:718 +0x165
k8s.io/kubernetes/vendor/k8s.io/klog/v2.FatalDepth(...)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/klog/v2/klog.go:1442
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/util.fatal(0xc0006a1500, 0x68, 0x1)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/util/helpers.go:93 +0x1f0
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/util.checkErr(0x1e5b480, 0xc0009b15e0, 0x1d06cf0)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/util/helpers.go:188 +0x945
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/util.CheckErr(...)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/util/helpers.go:115
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/version.NewCmdVersion.func1(0xc0009b3340, 0xc0006863f0, 0x0, 0x1)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/version/version.go:79 +0x117
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc0009b3340, 0xc0006863e0, 0x1, 0x1, 0xc0009b3340, 0xc0006863e0)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:846 +0x2c2
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0003a38c0, 0xc00007e180, 0xc00003a180, 0x3)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:950 +0x375
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:887
main.main()
        _output/dockerized/go/src/k8s.io/kubernetes/cmd/kubectl/kubectl.go:49 +0x21d

goroutine 6 [chan receive]:
k8s.io/kubernetes/vendor/k8s.io/klog/v2.(*loggingT).flushDaemon(0x2d04b80)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/klog/v2/klog.go:1131 +0x8b
created by k8s.io/kubernetes/vendor/k8s.io/klog/v2.init.0
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/klog/v2/klog.go:416 +0xd8

goroutine 11 [select]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x1d06c28, 0x1e5b780, 0xc0002f6000, 0x1, 0xc000042180)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:167 +0x149
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x1d06c28, 0x12a05f200, 0x0, 0x1, 0xc000042180)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0x1d06c28, 0x12a05f200, 0xc000042180)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/util/logs.InitLogs
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/util/logs/logs.go:51 +0x96
```

**Comment by stellirin** [comment]
I downgraded to `3.2.2` and functionality is restored:
```
kubectl --v=10 version
I0413 12:00:08.903388    1375 loader.go:375] Config loaded from file:  /home/vscode/.kube/config
I0413 12:00:08.906398    1375 round_trippers.go:424] curl -k -v -XGET  -H "User-Agent: kubectl/v1.19.7 (linux/amd64) kubernetes/1dd5338" -H "Accept: application/json, */*" 'https://34.71.63.69/version?timeout=32s'
I0413 12:00:09.389819    1375 loader.go:375] Config loaded from file:  /home/vscode/.kube/config
I0413 12:00:09.394554    1375 loader.go:375] Config loaded from file:  /home/vscode/.kube/config
I0413 12:00:09.399915    1375 loader.go:375] Config loaded from file:  /home/vscode/.kube/config
I0413 12:00:09.852162    1375 round_trippers.go:444] GET https://w.x.y.z/version?timeout=32s 200 OK in 946 milliseconds
I0413 12:00:09.852215    1375 round_trippers.go:450] Response Headers:
I0413 12:00:09.852222    1375 round_trippers.go:453]     Cache-Control: no-cache, private
I0413 12:00:09.852226    1375 round_trippers.go:453]     Content-Type: application/json
I0413 12:00:09.852229    1375 round_trippers.go:453]     Content-Length: 276
I0413 12:00:09.852231    1375 round_trippers.go:453]     Date: Tue, 13 Apr 2021 10:00:09 GMT
I0413 12:00:09.858364    1375 request.go:1097] Response Body: {
  "major": "1",
  "minor": "18+",
  "gitVersion": "v1.18.16-gke.302",
  "gitCommit": "2e4de00ee51e92a708578409838fd37044b00902",
  "gitTreeState": "clean",
  "buildDate": "2021-03-08T22:06:14Z",
  "goVersion": "go1.13.15b4",
  "compiler": "gc",
  "platform": "linux/amd64"
}
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.7", GitCommit:"1dd5338295409edcfff11505e7bb246f0d325d15", GitTreeState:"clean", BuildDate:"2021-01-13T13:23:52Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.16-gke.302", GitCommit:"2e4de00ee51e92a708578409838fd37044b00902", GitTreeState:"clean", BuildDate:"2021-03-08T22:06:14Z", GoVersion:"go1.13.15b4", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by ajoldham** [comment]
This is also happening on my 2017 Macbook 10.15.7... downgraded to 3.2.2 to fix.  Seems to break TLS to endpoints with self-generated certs?

Looking at 3.3.0 release notes wondering if this is breaking it?
- Apply domain-based HTTPS proxy no_proxy rules for encrypted connections. Fixes docker/for-mac#2732.


Was getting: 
`curl  -v https://192.168.1.1
*   Trying 192.168.1.1:443...
* TCP_NODELAY set
* Connected to 192.168.1.1 (192.168.1.1) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to 192.168.1.1:443
* Closing connection 0
curl: (35) OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to 192.168.1.1:443`

**Comment by rnorth** [comment]
@stellirin and @ajoldham are right, this is not just an M1 issue. Can we get this issue title and labels updated?

We're seeing this occur on Intel Macs, with containers trying to call HTTPS endpoints where the cert is (ultimately) signed by an internal root CA.

**Comment by randy5235** [comment]
Seeing a similar issue as well, running `openssl s_client -showcerts -connect stackoverflow.com:443 < /dev/null` would indicate a handshake failure and not show any certificates, rolling back to 3.2.2 resolved this issue. Running intel mpb here with 10.15.7

**Comment by i2rulloa** [comment]
Updated.

**Comment by randy5235** [comment]
Has anyone attempted this with the 3.3.1 update yet? I experience my issues with 3.3.0

**Comment by ajoldham** [comment]
Downloading the 3.3.1 "Mac with Intel Chip" from https://docs.docker.com/docker-for-mac/release-notes/ seems to be the same version as 3.3.0, and I didn't see any change in behavior.

**Comment by dotr** [comment]
Same SSL_ERROR_SYSCALL after update to 3.3.1, I can't access my kubernetes cluster from inside of the containers. After downgrade to 3.2.2 everything works ok, but Docker Desktop forces the upgrade to latest version and there'es no way to disable upgrades..

**Comment by Stono** [comment]
Aye also getting this on 3.3.1 when connecting to kubernetes masters with self-signed CA's.

**Comment by chucknelson** [comment]
Adding another data point just in case more comments = more attention :)

Same issue for 3.3.0 and 3.3.1 - and yeah, it only seems to be `https` sites with an entirely self-signed cert and/or cert chain, as @rnorth mentioned. If the top-level cert is a well known root cert, things seem fine.

In my company's case, this affects things like:
* Our internal Kubernetes calls (i.e., with `kubectl`) with servers that have just one self-signed cert.
* Outgoing traffic from our corporate proxy to public places like `google.com` where the proxy uses its own certs for the whole chain (i.e., the root cert is owned by my company).

**Comment by i2rulloa** [comment]
Found a workaround...

Run a socks5/http proxy on your bastion server or localhost (squidman). 
After that, `export HTTPS_PROXY=http://192.168.X.X:3333`

Enough for me in the meantime.

**Comment by navyzet** [comment]
My solution is downgrade

1) Uninstall docker from Docker Desktop 
![Снимок экрана 2021-04-27 в 14 49 04](https://user-images.githubusercontent.com/4208750/116236760-1d99fb00-a768-11eb-95ce-093c1cbc35f1.png)
2) Move Docker to trash from finder
3) install version 3.2.2
https://docs.docker.com/docker-for-mac/release-notes/#docker-desktop-322

Without the first item, the application gets stuck on a factory reset

**Comment by viyh** [comment]
I too had to downgrade to work around this. I couldn't use kubectl against GKE anymore from within a Docker container on my laptop, but downgrading fixed it.

**Comment by Anutrix** [comment]
> My solution is downgrade
> 
>     1. Uninstall docker from Docker Desktop
>        ![Снимок экрана 2021-04-27 в 14 49 04](https://user-images.githubusercontent.com/4208750/116236760-1d99fb00-a768-11eb-95ce-093c1cbc35f1.png)
> 
>     2. Move Docker to trash from finder
> 
>     3. install version 3.2.2
>        https://docs.docker.com/docker-for-mac/release-notes/#docker-desktop-322
> 
> 
> Without the first step, docker gets stuck on a factory reset

If someone is still facing factory reset stuck issue despite doing step 1, modify settingsVersion in `/Users/<username>/Library/Group Containers/group.com.docker/settings.json` to 6 before running Docker 3.2.2.

**Comment by navyzet** [comment]
I got the version automatically and ran into the problem again.
It's too bad that you can't turn off automatic updates.

**Comment by woxxy** [comment]
Because the autoupdate can't be disabled now I am stuck reinstalling every other time I start Docker for Mac. 

I suppose the autoupdate function is patched to ask in the later version, but because we have to rollback, we get pushed to the latest version every time.

**Comment by Baptiste-Villain** [comment]
Reinstalling 3.2.2 and going through this https://gist.github.com/LeZuse/aad7d02d2ccc564f7fc21f032607b47b has helped me workaround the auto update on restart.
Still hours of headache. Good luck guys!

**Comment by ddi-wtag** [comment]
Upgrading to 3.3.2 fixed it for me 🎉

**Comment by rune-bk** [comment]
Yea 3.3.2 seems to have fixed the proxy issue:

> - Disable the HTTP and HTTPS transparent proxies when there is no upstream proxy set. Fixes docker/for-mac#5572.
> - Revert to the HTTP and HTTPS proxy implementation used in 3.2.2.

**Comment by rohasplit** [comment]
> Upgrading to 3.3.2 fixed it for me



> Yea 3.3.2 seems to have fixed the proxy issue for me too:
> 
> > * Disable the HTTP and HTTPS transparent proxies when there is no upstream proxy set. Fixes #5572.
> > * Revert to the HTTP and HTTPS proxy implementation used in 3.2.2.

Did you folks do anything after the upgrade or did it work outright?

**Comment by lpfrenette** [comment]
> > Upgrading to 3.3.2 fixed it for me
> 
> > Yea 3.3.2 seems to have fixed the proxy issue for me too:
> > > * Disable the HTTP and HTTPS transparent proxies when there is no upstream proxy set. Fixes #5572.
> > > * Revert to the HTTP and HTTPS proxy implementation used in 3.2.2.
> 
> Did you folks do anything after the upgrade or did it work outright?

For me it worked right away.

**Comment by rune-bk** [comment]
> > Upgrading to 3.3.2 fixed it for me
> 
> > Yea 3.3.2 seems to have fixed the proxy issue for me too:
> > > * Disable the HTTP and HTTPS transparent proxies when there is no upstream proxy set. Fixes #5572.
> > > * Revert to the HTTP and HTTPS proxy implementation used in 3.2.2.
> 
> Did you folks do anything after the upgrade or did it work outright?

Worked right away. I upgraded from the broken version to 3.2.2 and it worked. No other changes. 
Had a very reproducible setup that suddenly worked

**Comment by jjqq2013** [comment]
is this issue concerned with the certificate? It seems not.

A single curl command to any ip in the VPN will show error 
>curl: (35) OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection

If this issue does not concern with the certificate, how about change the title of the issue?

**Comment by stephen-turner** [comment]
Closing based on comments in this ticket that it's fixed in 3.3.2.
